### ISSUE TITLE
Actions API fails randomly

### ISSUE THREAD

**Issue by brunocalou** [excaliburjs/Excalibur#3311]
### Steps to Reproduce

It's hard to reproduce the issue reliably, but it's happening pretty often. I'm using the actions api to render an impact animation to the screen. Here's my implementation:

```typescript
import { Actor, CollisionType, Color, type ActorArgs } from 'excalibur';
import { settings } from '../settings';

export class Impact extends Actor {
	constructor(config?: ActorArgs) {
		super({
			color: Color.White,
			opacity: 0,
			z: settings.z.ui,
			collisionType: CollisionType.PreventCollision,
			...config,
		});
	}

	blink() {
		this.actions.clearActions();
		this.actions.fade(0.8, 50).delay(20).fade(0, 100);
	}

	reset() {
		this.actions.clearActions();
		this.graphics.opacity = 0;
	}
}
```

Quite simple. But when I call the `blink` method, it randomly stops the animation. Here's a video of the issue:

https://github.com/user-attachments/assets/5d77d6ab-a54c-46d6-8eeb-4b08a9576790

~And here's a link to the website, if you need to reproduce the issue somehow: [Flappy Birdie](https://www.lofiandgames.com/flappy-birdie)~ Edit: I changed the implementation so it does not rely on the actions api

Because of this bug, I've added a `reset` method so even if the animation fails, I can stop showing it to the user.

I've tried to convert the actions to a promise and log the results, but it silently fails:

```typescript
this.actions
	.fade(0.8, 50)
	.delay(20)
	.fade(0, 100)
	.toPromise()
	.then(console.log)
	.catch(console.error);
```

### Expected Result

The actions should be executed in order

### Actual Result

The actions randomly fail to execute until the end

### Environment

- browsers and versions: Chrome, Safari
- operating system:  MacOS, iOS
- Excalibur versions: `^0.29.3`

### Current Workaround

~None, I clear the actions and change the opacity manually on a reset method, but it's suboptimal~

Since the animation is quite simple, I've changed the implementation to not rely on the Actions API:

```typescript
import { Actor, CollisionType, Color, Engine, type ActorArgs } from 'excalibur';
import { settings } from '../settings';

export class Impact extends Actor {
	private isBlinking = false;
	private time = 0;

	constructor(config?: ActorArgs) {
		super({
			color: Color.White,
			opacity: 0,
			z: settings.z.ui,
			collisionType: CollisionType.PreventCollision,
			...config,
		});
	}

	blink() {
		this.graphics.opacity = 0;
		this.time = 0;
		this.isBlinking = true;
	}

	reset() {
		this.isBlinking = false;
		this.graphics.opacity = 0;
		this.time = 0;
	}

	onInitialize(engine: Engine): void {
		engine.on('predraw', (event) => {
			if (this.isBlinking) {
				this.time += event.delta;

				if (this.time < 50) {
					this.graphics.opacity = Math.min(this.graphics.opacity + 0.2, 0.8);
				} else if (this.time > 70) {
					this.graphics.opacity = Math.max(0, this.graphics.opacity - 0.15);
				}
			}
		});
	}
}

```

**Comment by eonarheim** [comment]
@brunocalou thanks for the bug report! Definitely looks like a bug

**Comment by eonarheim** [comment]
I've done a little digging and I'm having trouble reproducing. I'm wonder if in the original example the blink method is called repeatedly by some other game logic. I've got some code that will appear to have the screen stuck white, but it's with the key held and the blink implementation called on repeat. If I comment out the `screenActor.actions.clearActions();` I get a strobe effect with the actions stacking up like I'd expect.


https://github.com/user-attachments/assets/5f2e1061-1583-42dc-b21c-6082be9249ff



```typescript

var engine = new ex.Engine({
  width: 800,
  height: 800
});


var screenActor = new ex.Actor({
  pos: ex.vec(0, 0),
  anchor: ex.vec(0, 0),
  width: engine.screen.width,
  height: engine.screen.height,
  color: ex.Color.White,
  opacity: 0
});

engine.add(screenActor);


engine.input.keyboard.on('hold', () => {
  screenActor.actions.clearActions();
  screenActor.actions
    .fade(0.8, 50)
    .delay(20)
    .fade(0, 100)
    .toPromise()
    .then(console.log)
    .catch(console.error);
});

engine.start();
```

**Comment by eonarheim** [comment]
I'll do some more digging, it might be a bug that has since been fixed on main from the 0.29.3 version...

**Comment by brunocalou** [comment]
I'm unable to reproduce the issue with the code I provided. I tried setting up a code pen and reintroducing the same code on the project, but the issue is not happening anymore. The lib version I was using was the `^0.29.3`.

I don't know exactly what was causing the problem, because the blink method was being called just once, but it seems to be some issue on my side. I remember fixing a memory leak this weekend, but I don't know if it's related.

Anyways, thanks for the help!
